actionpack/lib/action_dispatch/http/request.rb

```diff
@@ -319,7 +319,7 @@ def GET
     def POST
       @env["action_dispatch.request.request_parameters"] ||= Utils.deep_munge(normalize_encode_params(super || {}))
     rescue Rack::Utils::ParameterTypeError, Rack::Utils::InvalidParameterError => e
-      raise ActionController::BadRequest.new(:request, e)
+      raise ActionController::BadRequest, "Invalid request parameters: #{e.message}", e.backtrace
```

Comment:
agreed. I really want to see both the original exception and the last one on the debug exceptions page.